pkg/convert/pprof/streaming/labels_cache.go

```diff
@@ -0,0 +1,116 @@
+package streaming
+
+import (
+	"encoding/binary"
+	"github.com/cespare/xxhash/v2"
+	"github.com/pyroscope-io/pyroscope/pkg/storage/tree"
+	"sort"
+)
+
+type Labels []label
```

Comment:
Consider representing a label as a single uint64: we know for sure that both K and V << 2^32 so we can pack them `(k<<32)|v`. This reduces cost of:
1. sort
2. hashing <- I'm not even sure if we would need this: comparing underlying byte slices (of Labels) might be way cheaper

Reply:
Nice one.
I thought about another possible optimization: representing string as a uint64(pair of uint32, [from, to)) indices into profile buffer

Reply:
This should be also helpful. However, we still need to allocate a string for a reference (copy these bytes), otherwise memory ownership will be violated since the bytes are taken from pool. Therefore, depending on how many times a particular reference is resolved, it might even deteriorate the overall performance – though, I don't have a full picture and may be missing something